server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -1069,8 +1060,10 @@ public void updateCompactionStatus(TInfo tinfo, TCredentials credentials,
         case CANCELLED:
         case FAILED:
         case SUCCEEDED:
-          LONG_RUNNING_COMPACTIONS_BY_RG
-              .getOrDefault(rc.getGroupName(), new TimeOrderedRunningCompactionSet()).remove(rc);
+          var compactionSet = LONG_RUNNING_COMPACTIONS_BY_RG.get(rc.getGroupName());
+          if (compactionSet != null) {
+            compactionSet.remove(rc);
+          }
```

Comment:
This patterns avoids creating a `new TimeOrderedRunningCompactionSet()` then immediately calling `.remove()` on it